cocotb/__init__.py

```diff
@@ -92,6 +92,9 @@
     if not sys.warnoptions:
         warnings.simplefilter("default")
 
+InternalError = scheduler.InternalError
+"""InternalError should be accessible outside scheduler."""
```

Comment:
Not sure this is necessary. No one is going to able to catch this anyway, so there's little point making it easier to get to

Reply:
cocotb.scheduler is the instance of Scheduler defined in __init__.py.
Perhaps it makes more sense to rename scheduler.py to something else so we don't have this name collision?

Reply:
Not when you use the import keyword it isn't ;)

Reply:
Didn't think of that :).

Reply:
If we do `from cocotb.scheduler import InternalError' we get:
```ImportError: cannot import name 'InternalError' from partially initialized module 'cocotb.scheduler' (most likely due to a circular import) (/home/ben/Code/cocotb/cocotb/scheduler.py)```

Doing `from cocotb import scheduler` and then using `scheduler.InternalError` works fine.  But then we have `cocotb.scheduler` and `scheduler` both present in the decorators.py file, with one referring to the scheduler object, and the other referring to the module which is a bit smelly.